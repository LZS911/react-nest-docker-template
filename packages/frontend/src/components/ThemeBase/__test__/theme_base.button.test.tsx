import { render, renderHook } from '@testing-library/react';
import { IButtonProps } from '..';
import Button from '../Button/Button';
import useRenderButton from '../Button/useRenderButton';

type Props = Pick<IButtonProps, 'type' | 'className' | 'danger'>;
const initProps: Props = {};

describe('test useRenderButton', () => {
  test('should match snapshot', () => {
    const { result, rerender } = renderHook((v: Props = initProps) =>
      useRenderButton({ ...v })
    );
    expect(result.current).toMatchSnapshot();

    rerender({ type: 'primary' });
    expect(result.current).toMatchSnapshot();

    rerender({ type: 'secondary' });
    expect(result.current).toMatchSnapshot();

    rerender({ type: 'link' });
    expect(result.current).toMatchSnapshot();

    rerender({ type: 'primary', danger: true });
    expect(result.current).toMatchSnapshot();

    rerender({ type: 'secondary', danger: true });
    expect(result.current).toMatchSnapshot();

    rerender({ type: 'link', danger: true });
    expect(result.current).toMatchSnapshot();

    rerender({ type: 'default' });
    expect(result.current).toMatchSnapshot();

    rerender({ className: 'cls' });
    expect(result.current).toMatchSnapshot();

    rerender({ danger: true });
    expect(result.current).toMatchSnapshot();

    rerender({ danger: false });
    expect(result.current).toMatchSnapshot();
  });
});

describe('test Button', () => {
  test('should match snapshot', () => {
    const { container } = render(<Button />);

    expect(container).toMatchSnapshot();
  });
});
