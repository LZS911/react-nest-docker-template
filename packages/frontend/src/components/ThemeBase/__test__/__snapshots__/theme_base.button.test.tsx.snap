// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Button should match snapshot 1`] = `
<div>
  <button
    class="ant-btn ant-btn-default  dark:bg-darkMode focus:border-primary hover:border-primary focus:!text-primary hover:!text-primary"
    type="button"
  >
    <span
      class="dark:text-white hover:!text-primary focus:!text-primary"
    />
  </button>
</div>
`;

exports[`test useRenderButton should match snapshot 1`] = `
Object {
  "classNameList": Array [
    " dark:bg-darkMode focus:border-primary hover:border-primary focus:!text-primary hover:!text-primary",
    "dark:text-white hover:!text-primary focus:!text-primary",
  ],
  "realType": undefined,
}
`;

exports[`test useRenderButton should match snapshot 2`] = `
Object {
  "classNameList": Array [
    " border-primary focus:border-primary hover:border-primary !bg-primary",
    "!text-white",
  ],
  "realType": "primary",
}
`;

exports[`test useRenderButton should match snapshot 3`] = `
Object {
  "classNameList": Array [
    " border-secondary focus:border-secondary hover:border-secondary !bg-secondary",
    "!text-black dark:!text-white",
  ],
  "realType": undefined,
}
`;

exports[`test useRenderButton should match snapshot 4`] = `
Object {
  "classNameList": Array [
    undefined,
    "text-xs, text-gray-500, opacity-90, hover:text-sky-500",
  ],
  "realType": "link",
}
`;

exports[`test useRenderButton should match snapshot 5`] = `
Object {
  "classNameList": Array [
    "  !bg-primary",
    undefined,
  ],
  "realType": "primary",
}
`;

exports[`test useRenderButton should match snapshot 6`] = `
Object {
  "classNameList": Array [
    "  !bg-secondary",
    undefined,
  ],
  "realType": undefined,
}
`;

exports[`test useRenderButton should match snapshot 7`] = `
Object {
  "classNameList": Array [
    undefined,
    "",
  ],
  "realType": "link",
}
`;

exports[`test useRenderButton should match snapshot 8`] = `
Object {
  "classNameList": Array [
    " dark:bg-darkMode focus:border-primary hover:border-primary focus:!text-primary hover:!text-primary",
    "dark:text-white hover:!text-primary focus:!text-primary",
  ],
  "realType": "default",
}
`;

exports[`test useRenderButton should match snapshot 9`] = `
Object {
  "classNameList": Array [
    "cls dark:bg-darkMode focus:border-primary hover:border-primary focus:!text-primary hover:!text-primary",
    "dark:text-white hover:!text-primary focus:!text-primary",
  ],
  "realType": undefined,
}
`;

exports[`test useRenderButton should match snapshot 10`] = `
Object {
  "classNameList": Array [
    " dark:bg-darkMode ",
    undefined,
  ],
  "realType": undefined,
}
`;

exports[`test useRenderButton should match snapshot 11`] = `
Object {
  "classNameList": Array [
    " dark:bg-darkMode focus:border-primary hover:border-primary focus:!text-primary hover:!text-primary",
    "dark:text-white hover:!text-primary focus:!text-primary",
  ],
  "realType": undefined,
}
`;
